### PR TITLE
Adding a method to compute heat capacity ratio in generic property pack

### PR DESCRIPTION
**Pull request by nareshsusarla**
## Summary/Motivation:
Adding a method for computing heat capacity ratio in the generic property package and the EoS base class

## Changes proposed in this PR:
- added method to compute heat capacity ratio in eos_base.py `def heat_capacity_ratio(b, p)`
-added support for the same in generic_property.py

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/generic_models/properties/core/eos/eos_base.py [hunk 1]
```diff
@@ -67,6 +67,11 @@ def cv_mol_phase(b, p):
     def cv_mol_phase_comp(b, p, j):
         raise NotImplementedError(_msg(b, "cv_mol_phase_comp"))
 
+    @staticmethod
+    def heat_capacity_ratio(b, p):
```
Comment (andrewlee94) [inline idaes/generic_models/properties/core/eos/eos_base.py:71]: This should be `heat_capacity_ratio_phase`.

File: idaes/generic_models/properties/core/generic/generic_property.py [hunk 2]
```diff
@@ -873,6 +873,7 @@ def define_metadata(cls, obj):
              'cv_mol': {'method': '_cv_mol'},
              'cv_mol_phase': {'method': '_cv_mol_phase'},
              'cv_mol_phase_comp': {'method': '_cv_mol_phase_comp'},
+             'heat_capacity_ratio': {'method': '_heat_capacity_ratio'},
```
Comment (andrewlee94) [inline idaes/generic_models/properties/core/generic/generic_property.py:876]: This should be `heat_capacity_ratio_phase` also,

File: idaes/generic_models/properties/core/generic/generic_property.py [hunk 3]
```diff
@@ -1927,6 +1928,19 @@ def rule_cv_mol_phase_comp(b, p, j):
             self.del_component(self.cv_mol_phase_comp)
             raise
 
+    def _heat_capacity_ratio(self):
```
Comment (andrewlee94) [inline idaes/generic_models/properties/core/generic/generic_property.py:1931]: As above.

File: idaes/generic_models/properties/core/generic/generic_property.py [hunk 4]
```diff
@@ -1927,6 +1928,19 @@ def rule_cv_mol_phase_comp(b, p, j):
...
+        try:
+            def rule_heat_capacity_ratio(b, p):
+                p_config = b.params.get_phase(p).config
+                return p_config.equation_of_state.heat_capacity_ratio(b, p)
+            self.heat_capacity_ratio = Expression(
+                self.phase_list,
+                rule=rule_heat_capacity_ratio,
+                doc="Heat capacity ratio")
```
Comment (andrewlee94) [inline idaes/generic_models/properties/core/generic/generic_property.py:1939]: Add "by phase" or something here to indicate that this is indexed by phase.

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
A couple of comments on component naming, plus tests are needed to prove that this works. Updating the docs would also be a good idea.

**Review by eslickj — COMMENTED** [review]
Looks good to me.  Just add a test and I'll approve.  Hopefully there are existing heat capacity tests you can use to easily write a test.

**Review by eslickj — APPROVED** [review]

File: idaes/generic_models/properties/core/eos/tests/test_ideal.py [hunk 5]
```diff
@@ -192,6 +192,17 @@ def test_cv_mol_phase_comp(m):
...
 
 
+@pytest.mark.unit
+def test_heat_capacity_ratio_phase(m):
+
+    m.props[1].cp_mol_phase = Var(m.params.phase_list)
+    m.props[1].cv_mol_phase = Var(m.params.phase_list)
+
+    for p in m.params.phase_list:
+        assert str(Ideal.heat_capacity_ratio_phase(m.props[1], p)) == (
```
Comment (andrewlee94) [inline idaes/generic_models/properties/core/eos/tests/test_ideal.py:202]: Probably not for this PR, but there really should be a set of tests for EoSBase as well, which should include this.

**Review by andrewlee94 — APPROVED** [review]
This looks good now, but we should hold off merging until the release process is completed.

**Pull request merged**